### PR TITLE
Restructure reporting for performance

### LINKED ISSUE #1594: Large QPC deployment reports cause HTTP timeouts when retrieving reports
## Bug Description:
In testing Postgres, I discovered timeouts retrieving large reports from out satellite scans.  There are several factors that contributed to this.
- The satellite test machines we use were recently scaled up.  The Satellite 6.2 has 20K systems.  The Satellite 6.1 has 7K.  These machines now provide a performance test.
- We switched to Postgres.  Postgres is a little slower than SQLite.
- If we are going to merge 100 reports, it's likely that we will have a lot of systems in a single report.

A deployment report with 8K systems is 37.5 MB (uncompressed) and 7K compressed.  Additionally, we query the DB for the 8K systems and build the JSON.

### Expected behavior:
Reports can be retrieved quickly.

### Actual behavior:
Large reports cannot be retrieved.

### Steps to reproduce:
List clear steps to reproduce the bug.

___

## Acceptance Criteria:

- [ ] Verify large reports can be retrieved via REST API without a timeout

### PR DESCRIPTION
**Pull request by kholdaway**
Closes #1594

### CODE REVIEW

File: docs/swagger.yml [hunk 1]
```diff
@@ -1209,27 +1207,47 @@ definitions:
...
+          format: "int64"
+          description: "The fact collection identifier"
+          example: 15
         sources:
           type: "array"
           items:
             $ref: "#/definitions/FactCollectionSource"
           minItems: 1
-  FactCollectionOut:
+  FactCollectionIn:
```
Comment (abaiken) [inline docs/swagger.yml]: Not a blocker but we should consider making this consistent. ie. FactCollection & Details seem to be the same thing

File: docs/swagger.yml [hunk 2]
```diff
@@ -1209,27 +1207,47 @@ definitions:
         type: "string"
         description: "Deletion error message"
         example: "Failed to delete"
-  FactCollectionIn:
+  DeploymentReport:
+    properties:
+        report_id:
+          type: "integer"
+          format: "int64"
+          description: "The id associate with report"
```
[the remark this answers is not in the document]
Comment (kholdaway) [reply docs/swagger.yml]: Fixed

File: docs/swagger.yml [hunk 3]
```diff
@@ -946,7 +944,7 @@ paths:
           schema:
             type: "array"
             items:
-              $ref: "#/definitions/SystemReportOut"
+              $ref: "#/definitions/DeploymentReport"
```
[the remark this answers is not in the document]
Comment (kholdaway) [reply docs/swagger.yml]: Fixed

File: docs/swagger.yml [hunk 4]
```diff
@@ -1209,27 +1207,47 @@ definitions:
...
+  DetailsReport:
     type: "object"
     required:
+      - report_id
       - sources
     properties:
+        report_id:
+          type: "integer"
+          format: "int64"
+          description: "The fact collection identifier"
```
[the remark this answers is not in the document]
Comment (kholdaway) [reply docs/swagger.yml]: Fixed

File: docs/swagger.yml (hunk 1 above)
Comment (kholdaway) [reply docs/swagger.yml]: Done

File: quipucords/fingerprinter/task.py [hunk 5]
```diff
@@ -123,14 +126,30 @@ def run(self, manager_interrupt):
...
+        if not deployment_report:
+            deployment_report = DeploymentReport()
+            deployment_report.save()
+
+            # Set the report ids.  Right now they are deployemnt report id
+            # but they do not have to
+            deployment_report.report_id = deployment_report.id
+            deployment_report.save()
+
+            # Set the report ids.  Right now they are deployemnt report id
```
Comment (abaiken) [inline quipucords/fingerprinter/task.py]: This seems like a repeat comment, remove one of them and change `deployemnt` to `deployment`

File: quipucords/fingerprinter/task.py [hunk 6]
```diff
@@ -180,6 +199,10 @@ def _process_fact_collection(self, manager_interrupt, fact_collection):
         self.scan_task.log_message('START FINGERPRINT PERSISTENCE')
         status_count = 0
         total_count = len(fingerprints_list)
+        deployment_report = fact_collection.deployment_report
+        # deployment_report_json = []
```
Comment (abaiken) [inline quipucords/fingerprinter/task.py]: Seems like this should be removed

Comment (kholdaway) [reply quipucords/fingerprinter/task.py]: done

File: docs/swagger.yml [hunk 7]
```diff
@@ -918,7 +916,7 @@ paths:
         200:
           description: "Detailed report successfully retrieved"
           schema:
-            $ref: "#/definitions/FactCollectionOut"
+            $ref: "#/definitions/DetailsReport"
```
Comment (abaiken) [inline docs/swagger.yml]: Change this to DetailsReportOut possibly to keep the old structure of how FactCollection was

**Pull request merged**